src/mpid/ch4/netmod/ucx/ucx_am.h

```diff
@@ -75,6 +76,12 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_am_isend(int rank,
             /* cleanup preselected amtype to avoid problem with reused request */
             MPIDI_UCX_AMREQUEST(sreq, am_type_choice) = MPIDI_UCX_AMTYPE_NONE;
             break;
+        case MPIDI_UCX_AMTYPE_PIPELINE:
+            mpi_errno = MPIDI_UCX_do_am_isend_pipeline(rank, comm, handler_id, am_hdr, am_hdr_sz,
+                                                       data, count, datatype, sreq, false);
+            /* cleanup preselected amtype to avoid problem with reused request */
+            MPIDI_UCX_AMREQUEST(sreq, am_type_choice) = MPIDI_UCX_AMTYPE_NONE;
```

Comment:
Won't this break deferred requests? Can it be done at request free instead?

Reply:
This would not be a problem as the deferred request uses different states internally. The `am_type_choice` here is simply caching the AM type of choice if it was previously decided by `MPIDI_NM_am_check_eager`.